vignettes/advanced.Rmd

```diff
@@ -221,20 +112,174 @@ See `?ci` for more information on which information can be extracted from this f
+
+```{r, echo = FALSE, eval = TRUE, fig.align='center', dpi = 50}
+knitr::include_graphics("img/travis-job-id.png")
+```
+
+Rather than going to the web interface and getting the "Job ID" there, you can also do it from the command line.
+First, you need to query the overall "Build ID" of the run:
+
+```{r eval = FALSE}
+travis::travis_get_builds()
+```
+
+```{sh eval = FALSE}
+A collection of 25 Travis CI builds:
+- id: 416858821, number: 342, state: errored, duration: 3820, event_type: push, ...
+- id: 416834229, number: 341, state: errored, duration: 3738, event_type: push, ...
+- id: 416135954, number: 340, state: errored, duration: 3338, event_type: push, ...
+- id: 416107143, number: 339, state: errored, duration: 3670, event_type: push, ...
+- id: 416095854, number: 338, state: passed, duration: 3135, event_type: push, ...
+```
+
+Because every "build" has multiple "Jobs", you can now query the "Job ID".
+For example, when you know that you want to enter the latest build, you can combine both commands:
+
+```{r eval = FALSE}
+build_id <- travis::travis_get_builds()[[1]]$id
+travis::travis_get_jobs(build_id)
+```
+
+```{sh eval = FALSE}
+A collection of 7 Travis CI jobs:
+- id: 416858822, allow_failure: FALSE, number: 342.1, state: passed, started_at: 2018-08-16T15:02:52Z, ...
+- id: 416858823, allow_failure: FALSE, number: 342.2, state: passed, started_at: 2018-08-16T15:02:18Z, ...
+- id: 416858824, allow_failure: FALSE, number: 342.3, state: passed, started_at: 2018-08-16T15:02:43Z, ...
+- id: 416858825, allow_failure: FALSE, number: 342.4, state: passed, started_at: 2018-08-16T15:02:23Z, ...
+- id: 416858826, allow_failure: FALSE, number: 342.5, state: passed, started_at: 2018-08-16T15:02:23Z, ...
+```
 
-Afterwards, you can simply call `travis::travis_debug_job(<jobID>)` and the build will restart.
+Finally, the build can be entered by calling `travis::travis_debug_job()`:
 
-Wait a few minutes until the build arrives at the point when it prints you the `ssh` command you can use to log in.
+```{r eval = FALSE}
+job_id <- travis::travis_get_jobs(build_id)[[1]]$id
+travis::travis_debug_job(job_id)
+```
+
+```{sh eval = FALSE}
+Finished restarting debug job 416983664 from Travis CI.
+```
+
+Wait a few minutes until the build arrives at the point when it tells you the SSH command you can use to enter the build.
 It should look similar to `ssh ukjiuCEkxBBnRAe32Y8xCH0zj@ny2.tmate.io`.
+If you still do not want to use the browser manually, simple call `travis::browse()` to open your project on Travis.
+
+Once you are in the build you can use `bash` functions defined by Travis to to execute specific stages:
+
+```{sh eval = FALSE}
+travis_run_before_install
+travis_run_install
+travis_run_before_script
+travis_run_script
+travis_run_after_success
+travis_run_after_failure
+travis_run_after_script
+```
+
+The debug session is a VM running a [`tmate`](https://tmate.io/) session.
```

Comment:
Can we mention that `tmux` is running, and point to https://tmuxcheatsheet.com/?

Reply:
done in https://github.com/ropenscilabs/tic/pull/80/commits/c6507532ae6fc36f475f39264575253a01e01337